overlord/confdbstate/confdbstate.go

```diff
@@ -57,21 +61,35 @@ func SetViaView(bag confdb.Databag, view *confdb.View, requests map[string]inter
 }
 
 // GetView returns the view identified by the account, confdb schema and view name.
-func GetView(st *state.State, account, dbSchemaName, viewName string) (*confdb.View, error) {
-	confdbSchemaAs, err := assertstateConfdbSchema(st, account, dbSchemaName)
+func GetView(st *state.State, account, schemaName, viewName string) (*confdb.View, error) {
+	confdbSchemaAs, err := assertstateConfdbSchema(st, account, schemaName)
 	if err != nil {
-		if errors.Is(err, &asserts.NotFoundError{}) {
+		if !errors.Is(err, &asserts.NotFoundError{}) {
+			return nil, err
+		}
```

Comment:
should we do this if there is no custodian snap? or we will not get here in that case?